pkg/virt-handler/vm.go

```diff
@@ -331,6 +337,21 @@ func (d *VirtualMachineController) clearPodNetworkPhase1(vmi *v1.VirtualMachineI
 	defer d.phase1NetworkSetupCacheLock.Unlock()
 
 	delete(d.phase1NetworkSetupCache, vmi.UID)
+
+	// Clean Pod interface cache from map and files
+	d.podInterfaceCacheLock.Lock()
```

Comment:
Oh, one comment here that I missed. We need to address one thing here. It's super simple. Just add a unlock after line 339 for the phase1NetworkSetupCache lock instead of using the defer. 

It's dangerous to have nested locking. In this case I *think* we're fine, but we really don't want to go down this path. I'll explain

Nested locking establishes a locking order. So lets say we have a mutex A and B

```
lock A
lock B

some code

unlock B
unlock A
```

now if another piece of code does the following

```
lock B
lock A

some code

unlock A
unlockB
```

We have the potential for somethign called a deadlock. The first piece of code could have grabbed lock A, and the second could grab lock B in parallel and then neither of them can move forward.

Reply:
Good catch, corrected now